packages/connect-agreement/src/thegraph/connector.ts

```diff
@@ -0,0 +1,132 @@
+      throw new Error(
+        'AgreementConnectorTheGraph requires subgraphUrl to be passed.'
+      )
+    }
+    this.#gql = new GraphQLWrapper(subgraphUrl, verbose)
+  }
+
+  async disconnect() {
+    this.#gql.close()
+  }
+
+  async agreement(agreement: string): Promise<AgreementData> {
+    return this.#gql.performQueryWithParser(
+      queries.GET_AGREEMENT('query'),
+      { agreement },
+      (result: QueryResult) => parseAgreement(result)
+    )
+  }
+
+  async currentVersion(agreement: string): Promise<Version> {
+    return this.#gql.performQueryWithParser(
+      queries.GET_CURRENT_VERSION('query'),
+      { agreement },
+      (result: QueryResult) => parseCurrentVersion(result, this)
+    )
+  }
+
+  onCurrentVersion(agreement: string, callback: Function): SubscriptionHandler {
+    return this.#gql.subscribeToQueryWithParser(
+      queries.GET_CURRENT_VERSION('subscription'),
+      { agreement },
+      callback,
+      (result: QueryResult) => parseCurrentVersion(result, this)
+    )
+  }
+
+  async version(versionId: string): Promise<Version> {
+    return this.#gql.performQueryWithParser(
+      queries.GET_VERSION('query'),
+      { versionId },
+      (result: QueryResult) => parseVersion(result, this)
+    )
+  }
+
+  onVersion(versionId: string, callback: Function): SubscriptionHandler {
+    return this.#gql.subscribeToQueryWithParser(
+      queries.GET_VERSION('subscription'),
+      { versionId },
+      callback,
+      (result: QueryResult) => parseVersion(result, this)
+    )
+  }
+
+  async versions(agreement: string, first: number, skip: number): Promise<Version[]> {
+    return this.#gql.performQueryWithParser(
+      queries.ALL_VERSIONS('query'),
+      { agreement, first, skip },
+      (result: QueryResult) => parseVersions(result, this)
+    )
+  }
+
+  onVersions(agreement: string, callback: Function): SubscriptionHandler {
+    return this.#gql.subscribeToQueryWithParser(
+      queries.ALL_VERSIONS('subscription'),
+      { agreement, first: 1000, skip: 0 },
+      callback,
+      (result: QueryResult) => parseVersions(result, this)
+    )
+  }
+
+  async signer(signerId: string): Promise<Signer> {
+    return this.#gql.performQueryWithParser(
+      queries.GET_SIGNER('query'),
+      { signerId },
+      (result: QueryResult) => parseSigner(result, this)
+    )
+  }
+
+  onSigner(signerId: string, callback: Function): SubscriptionHandler {
+    return this.#gql.subscribeToQueryWithParser(
+      queries.GET_SIGNER('query'),
+      { signerId },
+      callback,
+      (result: QueryResult) => parseSigner(result, this)
+    )
+  }
+
+  async signatures(signerId: string, first: number, skip: number): Promise<Signature[]> {
+    return this.#gql.performQueryWithParser(
+      queries.GET_SIGNATURES('query'),
+      { signerId, first, skip },
+      (result: QueryResult) => parseSignatures(result, this)
+    )
+  }
+
+  onSignatures(signerId: string, callback: Function): SubscriptionHandler {
```

Comment:
Same than `onVersions()`: we should align the parameters with `signatures()`, and probably move `first` and `skip` in a filters object.